client/client-multi/client-c/src/kaa/platform-impl/posix/kaa_client.c

```diff
@@ -129,13 +130,20 @@ kaa_error_t on_kaa_tcp_channel_event(void *context, kaa_tcp_channel_event_t even
 
 kaa_error_t kaa_client_create(kaa_client_t **kaa_client, kaa_client_props_t *props)
 {
-    (void)props;
     KAA_RETURN_IF_NIL(kaa_client, KAA_ERR_BADPARAM);
-
+    kaa_error_t error_code;
+    if (props == NULL) {
+        error_code = kaa_default_working_directory_set();
```

Comment:
Remove this function. Use `kaa_client_props_set()`. all the necessary logic should be hidden there.